lambdas/sys_user_management/src/app.py

```diff
@@ -0,0 +1,151 @@
+
+
+    def sys_user_management(self, event):
+
+        names = event.get('names', [])
+
+        users = {}
+
+        for name in names:
+            password = self.generate_random_password()
+
+            self.iam_client.create_user(UserName=name)
+            self.iam_client.create_login_profile(UserName=name, Password=password, PasswordResetRequired=False)
+            self.iam_client.attach_user_policy(UserName=name, PolicyArn='arn:aws:iam::aws:policy/AdministratorAccess')
+            users[name] = password
+
+        logger.info(f"Created users: {json.dumps(users)}")
+        return users
+
+
+    def sys_user_management_cleanup(self):
+
+        self.delete_users_and_groups()
+        self.delete_customer_managed_policies()
+
+        regions = [region['RegionName'] for region in self.iam_client.describe_regions()['Regions']]
+        for region in regions:
+            self.delete_all_lambda_functions(region)
+
+        self.delete_iam_roles()
+
+
+    def delete_all_lambda_functions(self, region):
+        self.lambda_client = boto3.client('lambda', MasterRegion=region)
+        response = self.lambda_client.list_functions()
+
+        for function in response['Functions']:
+            function_name = function['FunctionName']
+            if function_name not in self.config['cleanup_exclude_list_lambda']:
+                self.lambda_client.delete_function(FunctionName=function_name)
+                logger.info('deleted lambda function', function_name)
+
+                self.stats['lambdas_deleted'] += 1
+                self.stats['lambdas_deleted'].append(function_name)
+
+
+    def delete_users_and_groups(self):
+
+        response = self.iam_client.list_users()
+        users = response['Users']
+
+        for user in users:
+            username = user['UserName']
+            user_groups = self.iam_client.list_groups(UserName=users['UserName'])
+
+            for group in user_groups:
+                self.iam_client.delete_group(GroupName=group)
+                self.stats['groups_deleted'] += 1
+                self.stats['groups_deleted'].append(group)
+
+            self.iam_client.delete_user(UserName=username)
+            self.iam_client.delete_login_profile(UserName=username)
+            logger.info('Removed user:', username)
+            self.stats['users_deleted'] += 1
+            self.stats['users_deleted'].append(username)
+
+
+    def delete_customer_managed_policies(self):
+
+        response = self.iam_client.list_policies(Scope='Local')
+        policies_to_die = response['PolicyName']
+
+        for policy in policies_to_die:
+            self.iam_client.delete_policy(PolicyName=policy)
+            logger.info("Deleted policy:", policy)
+            self.stats['policies_deleted'] += 1
+            self.stats['policies_deleted'].append(policy)
+
+
+    def delete_iam_roles(self):
+
+        response = self.iam_client.list_roles()
+        roles_to_die = response['Roles']
```

Comment:
`roles = [x for x in response['Roles'] if x not in self.config['cleanup_exclude_list_role']]`